storage/src/main/java/org/apache/kafka/server/log/remote/storage/ClassLoaderAction.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.log.remote.storage;
+
+/**
+ * This interface is used to execute any remote storage/metadata related operation.
+ *
+ * @param <T> return type for execute operation
+ * @param <E> Exception type to be thrown
+ */
+@FunctionalInterface
+public interface ClassLoaderAction<T, E extends Exception> {
```

Comment:
I hadn't realised this is used by multiple classes. Since it is, maybe we can move this to the `internals` package and call it simply `Action` or something like that. I can then use it within one of the index classes top.

Reply:
I set the name as `RemoteStorageAction` as it was meant for any remote storage actions. I though you wanted this to be classloader specific in your earlier comment. I am fine with renaming it to `StorageAction` so that it can be reused for any storage specific operations.

Reply:
Yeah, my original suggestion assumed it would be used within one class. I didn't realize we had multiple "ClassLoaderAware" classes. Once it's a top level interface, it seems better to keep it general. Thanks.